courses/serializers.py

```diff
@@ -152,6 +153,15 @@ def get_topics(self, instance):
             key=lambda topic: topic["name"],
         )
 
+    def get_page(self, instance):
+        return (
+            CoursePageSerializer(
+                instance=CoursePage.objects.filter(course=instance).get()
+            ).data
+            if CoursePage.objects.filter(course=instance).exists()
```

Comment:
We are making an extra query here. First, it checks for existence, and then it gets the object. `.first()` would have reduced this query to 1.